docs/spec.md

```diff
@@ -2202,9 +2202,11 @@ Encapsulates an implementation-defined operation `call_target_name` that takes
 `backend_config` and `api_version` may be used to provide additional
 implementation-defined metadata.
 
-At the moment, this operation contains a fairly disorganized collection of
-metadata which reflects organic evolution of its counterpart operation in
-the XLA compiler. In the future, we are planning to unify this metadata
+The metadata `has_side_effect` is used to prevent optimizations that can
+eliminate the op or execute it speculatively. The op has other metadata, like
```

Comment:
What does it mean to execute the op speculatively? 

Is this referring to reordering it with other ops? If that's the case, then this formulation looks inconsistent with the discussion in #1365: "The semantics of a true value of the proto field prevent optimizations like DCE, CSE etc on the side effecting custom_call. However, it does not impose any ordering restrictions. Two unknown side-effecting custom call nodes in HLO DAG without any explicit data dependency between them can be executed in any order".